src/Microsoft.Extensions.Hosting/Internal/ConsoleLifetime.cs

```diff
@@ -0,0 +1,52 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Threading;
+using System.Threading.Tasks;
+
+namespace Microsoft.Extensions.Hosting.Internal
+{
+    /// <summary>
+    /// Listens for Ctrl+C or SIGTERM and initiates shutdown.
+    /// </summary>
+    public class ConsoleLifetime : IHostLifetime
+    {
+        public ConsoleLifetime(IHostingEnvironment environment, IApplicationLifetime applicationLifetime)
+        {
+            Environment = environment ?? throw new ArgumentNullException(nameof(environment));
+            ApplicationLifetime = applicationLifetime ?? throw new ArgumentNullException(nameof(applicationLifetime));
+            ApplicationLifetime.ApplicationStarted.Register(() =>
+            {
+                Console.WriteLine("Application started. Press Ctrl+C to shut down.");
+                Console.WriteLine($"Hosting environment: {Environment.EnvironmentName}");
+                Console.WriteLine($"Content root path: {Environment.ContentRootPath}");
+            });
+        }
+
+        public IHostingEnvironment Environment { get; }
```

Comment:
Is there a need to expose this? It seems to be only used by the constructor here.